homeassistant/components/fritz/common.py

```diff
@@ -285,3 +306,111 @@ def device_info(self):
             "model": self._fritzbox_tools.model,
             "sw_version": self._fritzbox_tools.sw_version,
         }
+
+
+class FritzBoxBaseSwitch:
+    """Fritz switch base class."""
+
+    def __init__(self, fritzbox_tools: FritzBoxTools, switch_info: SwitchInfo) -> None:
+        """Init Fritzbox port switch."""
+        self.fritzbox_tools: FritzBoxTools = fritzbox_tools
+
+        self._description = switch_info["description"]
+        self._friendly_name = switch_info["friendly_name"]
+        self._icon = switch_info["icon"]
+        self._type = switch_info["type"]
+        self._update: Callable = switch_info["callback_update"]
+        self._switch: Callable = switch_info["callback_switch"]
+
+        self._name = f"{self._friendly_name} {self._description}"
+        self._unique_id = (
+            f"{self.fritzbox_tools.unique_id}-{slugify(self._description)}"
+        )
+
+        self._attributes: dict[str, str] = {}
+        self._is_available = True
+        self._last_toggle_timestamp: datetime | None = None
+
+        self._attr_is_on = False
+
+    @property
+    def name(self):
+        """Return name."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return name."""
+        return self._icon
+
+    @property
+    def unique_id(self):
+        """Return unique id."""
+        return self._unique_id
+
+    @property
+    def device_info(self):
+        """Return the device information."""
+
+        return {
+            "connections": {(CONNECTION_NETWORK_MAC, self.fritzbox_tools.mac)},
+            "identifiers": {(DOMAIN, self.fritzbox_tools.unique_id)},
+            "name": self._friendly_name,
+            "manufacturer": "AVM",
+            "model": self.fritzbox_tools.model,
+            "sw_version": self.fritzbox_tools.sw_version,
+        }
+
+    @property
+    def available(self) -> bool:
+        """Return availability."""
+        return self._is_available
+
+    @property
+    def device_state_attributes(self):
```

Comment:
Should be extra_state_attributes

Reply:
Here is an example: https://github.com/home-assistant/core/blob/5cc31a98e251f750e545b9936e3ceaa6ce7227cb/homeassistant/components/sonos/switch.py#L178-L190

Reply:
Done